libraries/aws/shadow/src/aws_iot_shadow_api.c

```diff
@@ -204,72 +204,71 @@ static AwsIotShadowError_t _validateThingNameFlags( _shadowOperationType_t type,
                                                     const AwsIotShadowCallbackInfo_t * pCallbackInfo,
                                                     const AwsIotShadowOperation_t * pOperation )
 {
-    AwsIotShadowError_t status = AWS_IOT_SHADOW_SUCCESS;
+    AwsIotShadowError_t status = AWS_IOT_SHADOW_BAD_PARAMETER;
 
     /* Type is not used when logging is disabled. */
     ( void ) type;
 
     /* Check Thing Name. */
-    if( AwsIot_ValidateThingName( pThingName, thingNameLength ) == false )
+    if( AwsIot_ValidateThingName( pThingName, thingNameLength ) == true )
```

Comment:
I'm not sure this function is more readable this way, as now both `AWS_IOT_SHADOW_BAD_PARAMETER` and `AWS_IOT_SHADOW_SUCCESS` are set at different parts, sometimes right after each other. It would be easier to follow if we initialize to one, and then set to the other.

Reply:
That makes sense. Let me try to refactor it.